core-bundle/src/Security/Authentication/AccessDecisionManager.php

```diff
@@ -12,34 +12,32 @@
 
 namespace Contao\CoreBundle\Security\Authentication;
 
-use Contao\CoreBundle\Routing\ScopeMatcher;
-use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Bundle\SecurityBundle\Security\FirewallContext;
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
 use Symfony\Component\Security\Core\Authorization\AccessDecisionManagerInterface;
 
 class AccessDecisionManager implements AccessDecisionManagerInterface
 {
     private AccessDecisionManagerInterface $inner;
     private AccessDecisionManagerInterface $contaoAccessDecisionManager;
-    private ScopeMatcher $scopeMatcher;
-    private RequestStack $requestStack;
+    private FirewallContext $firewallContext;
 
-    public function __construct(AccessDecisionManagerInterface $inner, AccessDecisionManagerInterface $contaoAccessDecisionManager, ScopeMatcher $scopeMatcher, RequestStack $requestStack)
+    public function __construct(AccessDecisionManagerInterface $inner, AccessDecisionManagerInterface $contaoAccessDecisionManager, FirewallContext $firewallContext)
```

Comment:
Because the constructor parameters changed, the services.yaml needs to be updated too: https://github.com/contao/contao/blob/001cea17fe99c1d2a44206c57333f49dfb594bae/core-bundle/src/Resources/config/services.yml#L710-L717

Reply:
Isn't this a BC break?

Reply:
Why? The constructor parameters of our services are not part of our BC promise AFAIK.

Reply:
You are right, according to our BC promise the constructor needs to be marked as `@internal` to allow that change: https://docs.contao.org/dev/internals/bc-promise/

Reply:
Added in 211066b26bfc4a0fb919feb754952f6d93768abc.